### PR TITLE
Game_event linkedTo addition

### PR DESCRIPTION
**Pull request by killerwife**
PRing this as per @X-Savior request

Adds the ability to link game_events

@Phatcat if you can, try to expedite this one, TBC-DB is being exported

### CODE REVIEW

**Comment by Phatcat** [comment]
Okay, I'll look at this asap.

**Comment by robinsch** [comment]
What is this for? I mean I get what it does.

File: src/game/GameEventMgr.cpp [hunk 1]
```diff
@@ -656,12 +680,17 @@ void GameEventMgr::UnApplyEvent(uint16 event_id)
 void GameEventMgr::ApplyNewEvent(uint16 event_id, bool resume)
 {
     m_ActiveEvents.insert(event_id);
+    sLog.outString("GameEvent %u \"%s\" started.", event_id, mGameEvent[event_id].description.c_str());
+    if (event_id==987)
```
Comment (robinsch) [inline src/game/GameEventMgr.cpp]: code format and magic numbers

Comment (killerwife) [inline src/game/GameEventMgr.cpp]: Daily restart, nuff said.

**Comment by killerwife** [comment]
Have an event like brewfest, or darkmoon faire, that has several smaller components, like, lets say, hourly events with some smaller shenanigans. You NEVER want this smaller event to be started when the bigger event is not started, lets say, because of spawns or maybe something else. This adds the ability to link events to each other, so that some cannot be started if others are not running.

File: src/game/GameEventMgr.cpp (hunk 1 above)
Comment (robinsch) [inline src/game/GameEventMgr.cpp]: Comment was missing before.

**Comment by robinsch** [comment]
Are you sure it's the right way to handle those smaller events with game_event? I mean even if you could simply add the start time, end time and change the occurence that would have the same effect.

**Comment by killerwife** [comment]
Well, I dont design game-events, and that wouldnt provide this failsafe. + I am not the author of this change, as it was done way before my time.

**Comment by killerwife** [comment]
@cala @xfurry @Grz3s Please review it :) since it is a DB change.

**Comment by Phatcat** [comment]
I see no problem with this, but then again, I am not big on game events, so I can't vouch for anything beyond the codestyle.

**Comment by dagochen** [comment]
I would do this with a different approach. We allready have conditions, so why not give an event a condition?

**Comment by Phatcat** [comment]
Wrath https://github.com/cmangos/mangos-wotlk/commit/7e23f3cc02cc4f8c339f973e1169d41c7eb30792
TBC https://github.com/cmangos/mangos-tbc/commit/e8771822dbd0b919001d2a574a0c72f35ce2c917
classic https://github.com/cmangos/mangos-classic/commit/b084f64d44342e75fd49f477bd677003cecebc20

**Pull request closed without merging**